sentry/src/main/java/io/sentry/transport/RateLimiter.java

```diff
@@ -113,6 +113,29 @@ public boolean isActiveForCategory(final @NotNull DataCategory dataCategory) {
     return false;
   }
 
+  @SuppressWarnings({"JdkObsolete", "JavaUtilDate"})
+  public boolean isAnyRateLimitActive() {
+    final Date currentDate = new Date(currentDateProvider.getCurrentTimeMillis());
+
+    // check all categories
+    final Date dateAllCategories = sentryRetryAfterLimit.get(DataCategory.All);
+    if (dateAllCategories != null) {
+      if (!currentDate.after(dateAllCategories)) {
+        return true;
+      }
+    }
+
+    for (DataCategory dataCategory : sentryRetryAfterLimit.keySet()) {
+      // check for specific dataCategory
+      final Date dateCategory = sentryRetryAfterLimit.get(dataCategory);
+      if (dateCategory != null) {
+        return !currentDate.after(dateCategory);
```

Comment:
this will return true/false on the first category regardless of result, you need the same as above
```suggestion
      if (!currentDate.after(dateCategory)) {
        return true;
      }
```

Reply:
ah oopsie, good catch